apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -690,19 +690,57 @@ namespace MWRender
         {
             mCamera->rotateCamera(-ptr.getRefData().getPosition().rot[0], -ptr.getRefData().getPosition().rot[2], false);
         }
+        if(!ptr.getRefData().isBaseNodeFlatten())
+        static_cast<SceneUtil::PositionAttitudeTransform*>(ptr.getRefData().getBaseNode())->setAttitude(rot);
+        else {
+            //need reflatten
 
-        ptr.getRefData().getBaseNode()->setAttitude(rot);
+            static_cast<SceneUtil::PositionAttitudeTransform*>(ptr.getRefData().getBaseNode()->getChild(0)->getUserData())->setAttitude(rot);
+             ptr.getRefData().flattenTransform();
+
+        }
     }
 
     void RenderingManager::moveObject(const MWWorld::Ptr &ptr, const osg::Vec3f &pos)
     {
-        ptr.getRefData().getBaseNode()->setPosition(pos);
+
+        SceneUtil::getCellOrigin(ptr.getCell()->getCell());
+
+        osg::Vec3 cellorigin(0,0,0);
+        SceneUtil::PositionAttitudeTransform* trans;
+        if(ptr.getRefData().isBaseNodeFlatten())
+            trans=static_cast<SceneUtil::PositionAttitudeTransform*>(ptr.getRefData().getBaseNode()->getChild(0)->getUserData());
+        else trans=static_cast<SceneUtil::PositionAttitudeTransform*>(ptr.getRefData().getBaseNode());
+
+;
+        SceneUtil::PositionAttitudeTransform* ptrans;
+        if(ptr.getRefData().getBaseNode()->getNumParents()>0)
+        {
+            ptrans=dynamic_cast<SceneUtil::PositionAttitudeTransform*>(ptr.getRefData().getBaseNode()->getParent(0));
+            if(ptrans)
+                cellorigin=SceneUtil::getCellOrigin(ptr.getCell()->getCell());
+            else
+            {
+                if(!ptr.getClass().isNpc())
+                    OSG_WARN<<"fuck"<<std::endl;
```

Comment:
strange stuff here

Reply:
I've fixed this problem in https://github.com/OpenMW/openmw/pull/2421/commits/d4e0663ece86c6e51c41b422303ab0336c61931d
removing 
mObjectRoot->getParent(0)->removeChild(mObjectRoot);
in MWRender::Animation::setObjectRoot
Hope it doesn't mess with the previous purpose of that code

Reply:
ok it's the player that is initialize before any cell loading....perhaps player should be added in the scenegraph after cell loading